src/Rules/TooWideTypehints/TooWideMethodReturnTypehintRule.php

```diff
@@ -82,9 +82,9 @@ public function processNode(Node $node, Scope $scope): array
 
 		$returnType = TypeCombinator::union(...$returnTypes);
 		if (
-			!$method->isPrivate()
-			&& ($returnType->isNull()->yes() || $returnType instanceof ConstantBooleanType)
-			&& !$isFirstDeclaration
+			!$isFirstDeclaration
+			&& !$method->isPrivate()
+			&& ($returnType->isNull()->yes() || $returnType->isTrue()->yes() || $returnType->isFalse()->yes())
```

Comment:
Logical operators are short-circuit evaluated, so rearranging expressions can lead to micro-optimizations.